src/extension/app/components/modal/modal-container.js

```diff
@@ -155,6 +172,39 @@ export class ModalContainer extends LitElement {
     this.remove();
   }
 
+  /**
+   * Returns the bulk modal options.
+   * @returns {Object} The bulk modal options
+   */
+  getBulkModalOptions() {
+    const { bulkStore } = this.appStore;
+    if (!bulkStore || !bulkStore.summary) {
+      return undefined;
+    }
+
+    const { operation, resources, host } = bulkStore.summary;
+    const failed = resources.filter((resource) => resource.status >= 400);
+    const succeeded = resources.filter(({ status }) => status < 400);
+    const paths = succeeded.map(({ path }) => path);
+    const openUrlsLabel = this.appStore.i18n(`open_url${paths.length !== 1 ? 's' : ''}`)
+      .replace('$1', `${paths.length}`);
+    const copyUrlsLabel = this.appStore.i18n(`copy_url${paths.length !== 1 ? 's' : ''}`)
+      .replace('$1', `${paths.length}`);
+
+    const options = {};
+    options.underlay = true;
+    options.headline = bulkStore.getSummaryText(operation, resources.length, failed.length);
+    if (paths.length > 0) {
+      options.confirmLabel = openUrlsLabel;
+      options.confirmCallback = () => bulkStore.openUrls(host, paths);
+      options.secondaryLabel = copyUrlsLabel;
+      options.secondaryCallback = () => bulkStore.copyUrls(host, paths);
+    }
+    options.cancelLabel = this.appStore.i18n('close');
+    options.content = html`<bulk-result></bulk-result>`;
+    return options;
+  }
+
```

Comment:
@dylandepass I was struggling to extract the bulk modal into its own component, but isolated it to this method now. The tests for it are already in a [separate file](https://github.com/adobe/aem-sidekick/pull/152/files#diff-be303cefc5537964d526584694efdb554758e1b18382ea0cf78ca241d9c969e0). I [created an issue](#183) to fully separate the bulk modal in an upcoming PR.